pandas/core/generic.py

```diff
@@ -5201,60 +5201,217 @@ def abs(self):
+              objects submit the ``numpy.object`` data type. Strings
+              can also be used in the style of
+              ``select_dtypes`` (e.g. ``df.describe(include=['O'])``)
+        exclude : None (default) or a list of dtypes or strings, optional,
+            A black list of data types to omit from the result. Ignored
+            for Series. Here are the options:
+
+            - None (default). The result will exclude nothing.
+            - A list of dtypes or strings. Excludes the provided data types
+              from the result. To select numeric types submit
+              ``numpy.number``. To select categorical objects submit the data
+              type ``numpy.object``. Strings can also be used in the style of
+              ``select_dtypes`` (e.g. ``df.describe(include=['O'])``)
 
         Returns
         -------
-        summary: %(klass)s of summary statistics
+        summary: NDFrame of summary statistics
 
         Notes
         -----
-        The output DataFrame index depends on the requested dtypes:
 
-        For numeric dtypes, it will include: count, mean, std, min,
-        max, and lower, 50, and upper percentiles.
+        Analyzes both numeric and object series, as well
+        as DataFrame column sets of mixed data types.
 
-        For object dtypes (e.g. timestamps or strings), the index
-        will include the count, unique, most common, and frequency of the
-        most common. Timestamps also include the first and last items.
+        For numeric data, the result's index will include ``count``,
+        ``mean``, ``std``, ``min``, ``max`` as well as lower, ``50`` and
+        upper percentiles. By default the lower percentile is ``25`` and the
+        upper percentile is ``75``. The ``50`` percentile is typically the
+        same as the median.
 
-        For mixed dtypes, the index will be the union of the corresponding
-        output types. Non-applicable entries will be filled with NaN.
-        Note that mixed-dtype outputs can only be returned from mixed-dtype
-        inputs and appropriate use of the include/exclude arguments.
+        For object data (e.g. strings or timestamps), the result's index
+        will include ``count``, ``unique``, ``top``, and ``freq``. The ``top``
+        is the most common value. The ``freq`` is the most common value's
+        frequency. Timestamps also include the ``first`` and ``last`` items.
 
-        If multiple values have the highest count, then the
-        `count` and `most common` pair will be arbitrarily chosen from
+        If multiple object values have the highest count, then the
+        ``count`` and ``top`` results will be arbitrarily chosen from
         among those with the highest count.
 
-        The include, exclude arguments are ignored for Series.
+        For mixed data types provided via a DataFrame, the result will
+        include a union of attributes of each type.
```

Comment:
Maybe mention first that the default is only to show numeric columns. And if, using `include='all'`, different types are included, then the index is union of the attributes of each type.